stl/inc/ranges

```diff
@@ -6837,6 +6837,602 @@ namespace ranges {
+            return (_Lhs._M_Current <=> _Rhs._M_Current);
+        }
+
+        _NODISCARD_FRIEND constexpr _Iterator operator+(const _Iterator& _Lhs, difference_type _Rhs) noexcept(
+            noexcept(_STD declval<_Iterator>() += _Rhs) && is_nothrow_copy_assignable_v<_Iterator>) // strengthened
+            requires _Zip::_All_random_access<_IsConst, _ViewTypes...> {
+            auto _Modified_iterator = _Lhs;
+            _Modified_iterator += _Rhs;
+
+            return _Modified_iterator;
+        }
+
+        _NODISCARD_FRIEND constexpr _Iterator operator+(difference_type _Lhs, const _Iterator& _Rhs) noexcept(
+            noexcept(_Rhs + _Lhs)) /* strengthened */ requires _Zip::_All_random_access<_IsConst, _ViewTypes...> {
+            return (_Rhs + _Lhs);
+        }
+
+        _NODISCARD_FRIEND constexpr _Iterator operator-(const _Iterator& _Lhs, difference_type _Rhs) noexcept(
+            noexcept(_STD declval<_Iterator>() -= _Rhs) && is_nothrow_copy_assignable_v<_Iterator>) // strengthened
+            requires _Zip::_All_random_access<_IsConst, _ViewTypes...> {
+            auto _Modified_iterator = _Lhs;
+            _Modified_iterator -= _Rhs;
+
+            return _Modified_iterator;
+        }
+
+        _NODISCARD_FRIEND constexpr difference_type operator-(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
+            noexcept(_Zip::_Get_smallest_iterator_sentinel_distance<difference_type>(
+                _Lhs._M_Current, _Rhs._M_Current))) // strengthened
+            requires(sized_sentinel_for<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>,
+                iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>&&...) {
+            return _Zip::_Get_smallest_iterator_sentinel_distance<difference_type>(_Lhs._M_Current, _Rhs._M_Current);
+        }
+
+        _NODISCARD_FRIEND constexpr auto iter_move(const _Iterator& _Itr) noexcept(
+            (noexcept(_RANGES iter_move(_STD declval<const iterator_t<_Maybe_const<_IsConst, _ViewTypes>>&>())) && ...)
+            && (is_nothrow_move_constructible_v<range_rvalue_reference_t<_Maybe_const<_IsConst, _ViewTypes>>> && ...)) {
+            return _Zip::_Tuple_transform(_RANGES iter_move, _Itr._M_Current);
+        }
+
+        friend constexpr void iter_swap(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
+            (noexcept(iter_swap(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>(),
+                 _STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>()))
+                && ...)) requires(indirectly_swappable<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>&&...) {
+            const auto _Swap_every_pair_lambda = [&_Lhs, &_Rhs ]<size_t... _Indices>(index_sequence<_Indices...>) {
+                ((_RANGES iter_swap(_STD get<_Indices>(_Lhs._M_Current), _STD get<_Indices>(_Rhs._M_Current))), ...);
+            };
+
+            constexpr auto _Index_sequence = index_sequence_for<_ViewTypes...>{};

```

Comment:
Another occurrence where `auto` should be avoided. (It would also be reasonable to simply construct a temporary on the next line, your choice.)